tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -1297,6 +1297,76 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the TL-WPA8630P v2 (DE)*/
+	{
+		.id     = "TLWPA8630PV2DE",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:TL-WPA8630P,product_ver:2.0.0,special_id:44450000}\n",
+		.support_trail = '\x00',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"factory-boot", 0x00000, 0x20000},
+			{"fs-uboot", 0x20000, 0x20000},
+			{"firmware", 0x40000, 0x5e0000},
+			{"partition-table", 0x620000, 0x02000},
+			{"extra-para", 0x632100, 0x01000},
+			{"soft-version", 0x640000, 0x01000},
+			{"support-list", 0x641000, 0x01000},
+			{"profile", 0x642000, 0x08000},
+			{"user-config", 0x650000, 0x10000},
+			{"default-config", 0x660000, 0x10000},
+			{"default-nvm", 0x670000, 0xc0000},
+			{"default-pib", 0x730000, 0x40000},
+			{"default-mac", 0x7e0000, 0x00020},
+			{"pin", 0x7e0100, 0x00020},
+			{"device-id", 0x7e0200, 0x00030},
+			{"product-info", 0x7e1100, 0x01000},
+			{"radio", 0x7f0000, 0x10000},
+			{NULL, 0, 0}
+		},
+
+		.first_sysupgrade_partition = "os-image",
+		.last_sysupgrade_partition = "file-system"
+	},
+
+	/** Firmware layout for the TL-WPA8630P v2 (EU)*/
+	{
+		.id     = "TLWPA8630PV2EU",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:TL-WPA8630P,product_ver:2.0.0,special_id:45550000}\n",
+		.support_trail = '\x00',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"factory-boot", 0x00000, 0x20000},
```

Comment:
Should this be `factory-uboot`?

Tested on stock TL-WPA8630P v2 EU, special_id:45550000.

When this entry is `factory-boot`, 0x0-0x20000 is cleared after upgrading with `*-squashfs-factory.bin` through the stock web UI firmware update page, and the device is unable to boot - similar to [your report](https://forum.openwrt.org/t/sysupgrade-question-for-new-device-tl-wpa8630p-v2/38956/17)). When its changed to `factory-uboot`, upgrading from stock to OpenWRT works great. You can see the same spelling of `factory-uboot` in the partition labels section in the header of the official firmware. (Is it expected that unlabeled/mislabeled partitions get cleared by the TP-Link upgrade process?).

Reply:
Good catch! You have no idea how often I checked all the numbers to the right, but I missed the labels!
I checked different firmware versions, they all have the name set to `factory-uboot` and not `factory-boot`.

I hope I'll find the time to flash you backup soon, then I will test this and update the commit.Then, we've got at least the EU version verified!